### ISSUE TITLE
API reports wrong plugin type for homebridge-nukiio

### ISSUE THREAD

**Issue by BobbySlope** [hoobs-org/HOOBS#308]
https://www.npmjs.com/package/homebridge-nukiio
https://www.npmjs.com/package/homebridge-pushover

https://www.npmjs.com/package/homebridge-http-webhooks

**Comment by BobbySlope** [comment]
Was an combined issue together witch homebridge-pushover

config that works for home bridge-pushover:
{
    "platform": "Pushover",
    "name": "Pushover",
    "pushover": {
        "user": "Enter User",
        "token": "Enter Token"
    },
    "messages": [
        {
            "name": "Enter Name of the Switch",
            "message": "Enter Message",
            "priority": "high"
        },
        {
            "name": "Enter Name of the Switch",
            "message": "Enter Message",
            "priority": "high"
        },
        {
            "name": "Enter Name of the Switch",
            "message": "Enter Message"
        }
    ]
}

config that works for NukiIO:

Config needs to be done in advanced mode not in the plugin config section. because plugins config section shows accessory instead of platform

{
            "platform": "NukiBridge",
            "bridge_url": "http://ip-of-the-bridge:8080",
            "api_token": "apitoken",
            "request_timeout_lockstate": 15000,
            "request_timeout_lockaction": 45000,
            "request_timeout_other": 15000,
            "cache_directory": "./.node-persist/storage",
            "webhook_server_ip_or_name": "ip-of-the-webhook-server",
            "webhook_port": 51830,
            "lock_state_mode": 1,
            "lockaction_maxtries": 3,
            "lockaction_retrydelay": 30000,
            "add_maintainance_buttons": false,
            "locks": [
                {
                    "id": "nuki-id",
                    "name": "name",
                    "usesDoorLatch": true,
                    "priority": 1
                }
            ],
            "openers": [
                {
                    "id": "nuki-id",
                    "name": "name",
                    "priority": 1
                }
            ]
        }

**Comment by rpolenthon** [comment]
The API reports the wrong type for the plugin. That’s why the interface want configure it as accessory. 

I tested it in the app, same behavior.

**Comment by rpolenthon** [comment]
Another plugin was reported by a user: https://www.npmjs.com/package/homebridge-http-webhooks

It reports the accessory type instead of the platform type.

**Comment by mkellsy** [comment]
Both homebridge-nukiio and homebridge-http-webhooks have multiple register calls.

```
module.exports = function(homebridge) {
    homebridge.registerPlatform("homebridge-nukiio", "NukiBridge", NukiBridgePlatform);
    homebridge.registerAccessory("homebridge-nukiio", "NukiLock", NukiLockAccessory);
    homebridge.registerAccessory("homebridge-nukiio", "NukiOpener", NukiOpenerAccessory);
    homebridge.registerAccessory("homebridge-nukiio", "NukiBridgeMaintainanceSwitch", NukiBridgeMaintainanceSwitchAccessory);
};
```

homebridge-pushover is using `process.env.HOME` then `.homebridge` to access a special file. They should be using the storage path not enc.HOME.

**Comment by mkellsy** [comment]
I created an issue for homebridge-pushover.
https://github.com/meg768/homebridge-pushover/issues/7

**Comment by mkellsy** [comment]
I am turning the plugin type logic into an array that holds the paring of the type, alias and path. I have to do this because in the above example there are 4 different register calls with 4 different alias.

**Comment by mkellsy** [comment]
homebridge-pushover updated their code. It now works.

**Comment by mkellsy** [comment]
Fixed in 3.1.2
